lm_eval/models/huggingface.py

```diff
@@ -402,11 +402,97 @@ def world_size(self):
     def tokenizer_name(self) -> str:
         return self.tokenizer.name_or_path.replace("/", "__")
 
-    @property
-    def chat_template(self) -> str:
-        if self.tokenizer.chat_template is not None:
-            return self.tokenizer.chat_template
-        return self.tokenizer.default_chat_template
+    def chat_template(self, chat_template: Union[bool, str] = False) -> Optional[str]:
```

Comment:
It's confusing when reading this fn that `chat_template` is both the name of the return value from this function and the kwarg input. Could we change the name of the return value variable perhaps?